### PR TITLE
[Test] Introduce TestBuilder

### PR DESCRIPTION
**Pull request by dayo09**
This commit introduces 'TestBuilder', which is a helper class
to build a test file and directories during mocha test.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

### CODE REVIEW

File: src/Tests/OneExplorer/OneExplorer.test.ts [hunk 1]
```diff
@@ -15,141 +15,23 @@
...
-    }
-  }
-}
+import {TestBuilder} from '../TestBuilder';
 
 suite('OneExplorer', function() {
-  setup(() => TestBuilder.setUp());
-  teardown(() => TestBuilder.tearDown());
-  suite('OneExplorer', function() {
+  suite('OneExplorer2', function() {
```
Comment (dayo09) [inline src/Tests/OneExplorer/OneExplorer.test.ts:24]:
```suggestion
  suite('OneExplorer', function() {
```

File: src/Tests/TestBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      fs.rmdirSync(TestBuilder.tempDir, {recursive: true});
+      console.log(`Test directory is removed successfully. (${TestBuilder.tempDir})`);
+      fs.rmdirSync(TestBuilder.workspaceDir, {recursive: true});
+      console.log(
+          `Test directory in worksp is removed successfully. (${TestBuilder.workspaceDir})`);
+    } catch (e) {
+      // Do not throw to proceed the test
+      console.error('Cannot remove the test directory');
+    } finally {
+      console.log(`Removed ${TestBuilder.tempDir} and ${TestBuilder.tempDir}.`);
```
Comment (llFreetimell) [inline src/Tests/TestBuilder.ts:132]: These seems not matched each other :(
If an error is caught in `catch`, it means that the directories are not removed.

Comment (dayo09) [reply src/Tests/TestBuilder.ts:132]: Oh, sure. Thanks.

File: src/Tests/TestBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
```
Comment (dayo09) [inline src/Tests/TestBuilder.ts:133]:
```suggestion
    }
```

File: src/Tests/TestBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      }
+      if (fs.existsSync(TestBuilder.workspaceDir)) {
+        fs.rmdirSync(TestBuilder.workspaceDir, {recursive: true});
+      }
+
+      fs.mkdirSync(TestBuilder.tempDir, {recursive: true});
+      fs.mkdirSync(TestBuilder.workspaceDir, {recursive: true});
+    } catch (e) {
+      console.error('Cannot create temporal directory for the test');
+      throw e;
```
Comment (dayo09) [inline src/Tests/TestBuilder.ts:117]:
```suggestion
      // Do not throw to proceed the test
      console.error('Cannot create temporal directory for the test');
```

**Comment by gandi0330** [comment]
👍

**Review by gandi0330 — APPROVED** [review]

**Review by llFreetimell — APPROVED** [review]

**Pull request merged**